sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -114,14 +114,14 @@ class SessionCatalog(
    * Format table name, taking into account case sensitivity.
    */
   protected[this] def formatTableName(name: String): String = {
-    if (conf.caseSensitiveAnalysis) name else name.toLowerCase
+    if (conf.caseSensitiveAnalysis) name else name.toLowerCase(Locale.ROOT)
```

Comment:
We have the restrictions on database/table names. That is, the names only contain characters, numbers, and _. 

Without the fixe in this PR, users are not allowed to read/write/create a table whose name containing `I`, because `toLowerCase` will convert it to `ı` when the locale is `tr`. The names become illegal. Is my understanding right?